client/src/components/MapGeocoderWrapper.vue

```diff
@@ -29,20 +30,27 @@ import mapboxgl from 'mapbox-gl';
  */
 export default defineComponent({
   name: 'MapGeocoderWrapper',
-  setup() {
+  setup(props, { emit }) {
+    const visible = computed({
+      get: () => props.modelValue,
+      set: (value) => emit('update:modelValue', value),
+    });
     const state: State = inject(stateKey, State.default());
 
     return {
       state,
+      visible,
     };
   },
   data() {
     return {
       // Default to empty geocoder instance that is not connected to map or view.
       geocoder: new MapboxGeocoder(),
-      visible: false,
     };
   },
+  props: {
+    'modelValue': { type: Boolean, default: false },
```

Comment:
What does `modelValue` mean? The name makes me think it'd be a value from the model directory, but it's a bool.

Reply:
It's the v-model for this component. I did a custom name for clarity